spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -423,6 +454,55 @@ public boolean matchHeader(String headerName) {
 			return false;
 		}
 
+		@Override
+		public boolean isNegated() {
+			return false;
+		}
+
+	}
+
+	/**
+	 * A pattern-based {@link HeaderMatcher} that matches if the specified
+	 * header matches the specified simple pattern.
+	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
+	 * @since 4.3
+	 */
+	protected static class SinglePatternBasedHeaderMatcher implements HeaderMatcher {
```

Comment:
Cool! That's really better to avoid any breaking changes, plus the collection style may still have sense somewhere.
So, I found this `SinglePatternBasedHeaderMatcher` as the best one.
:+1: